front/app/containers/Admin/pagesAndMenu/containers/GenericHeroBannerForm/BannerImageFields/index.tsx

```diff
@@ -59,72 +67,95 @@ const BannerImageField = ({
   headerBg,
   onAddImage,
   onRemoveImage,
-  setFormStatus,
   onOverlayChange,
 }: Props) => {
   const { formatMessage } = useIntl();
   const [previewDevice, setPreviewDevice] = useState<TPreviewDevice>('desktop');
   const [headerLocalDisplayImage, setHeaderLocalDisplayImage] =
     useState<TLocalHeaderImage>(null);
   const [bannerError, setBannerError] = useState<TBannerError>(null);
-  useEffect(() => {
-    // the image file sent from the API needs to be converted
-    // to a format that can be displayed. this is done locally
-    // when the image is changed but needs to be done manually
-    // to process the initial API response
-    const convertHeaderToUploadFile = async (fileInfo) => {
-      if (fileInfo) {
-        const tenantHeaderBg = await convertUrlToUploadFile(fileInfo);
-        const headerBgUploadFile = !isNilOrError(tenantHeaderBg)
-          ? [tenantHeaderBg]
-          : [];
-        setHeaderLocalDisplayImage(headerBgUploadFile);
-        setBannerError(null);
-      }
-    };
 
+  useEffect(() => {
     const headerFileInfo = headerBg?.large;
+
     convertHeaderToUploadFile(headerFileInfo);
   }, [headerBg]);
 
-  // set error and disable save button if header is removed,
-  // the form cannot be saved without an image
   useEffect(() => {
+    // set error and disable save button if header is removed,
+    // the form cannot be saved without an image
     if (isNil(headerBg)) {
       setBannerError(formatMessage(messages.noHeader));
       return;
     }
+  }, [headerBg, formatMessage]);
 
-    setBannerError(null);
-  }, [headerBg, formatMessage, setFormStatus]);
+  const convertHeaderToUploadFile = async (
+    fileInfo: string | null | undefined
+  ) => {
+    // the image file sent from the API needs to be converted
+    // to a format that can be displayed. this is done locally
+    // when the image is changed but needs to be done manually
+    // to process the initial API response
+    if (fileInfo) {
+      const tenantHeaderBg = await convertUrlToUploadFile(fileInfo);
+
+      setHeaderLocalDisplayImage(
+        !isNilOrError(tenantHeaderBg) ? tenantHeaderBg : null
+      );
+    }
+  };
 
-  const handleOnAddImageToUploader = (newImage: UploadFile[]) => {
+  const handleOnAddImageToUploader = (newImages: UploadFile[]) => {
     // this base64 value is sent to the API
-    onAddImage(newImage[0].base64);
+    onAddImage(newImages[0].base64);
     // this value is used for local display
-    setHeaderLocalDisplayImage([newImage[0]]);
+    setHeaderLocalDisplayImage(newImages[0]);
+    setBannerError(null);
   };
 
   const handleOnRemoveImageFromUploader = () => {
     onRemoveImage();
     setHeaderLocalDisplayImage(null);
   };
 
-  const imageIsSaved = headerLocalDisplayImage?.[0].remote || false;
   const hasLocalHeaderImage = !isNilOrError(headerLocalDisplayImage);
+  const imageShouldBeSaved = headerLocalDisplayImage
+    ? headerLocalDisplayImage.remote
```

Comment:
This needs to be `!headerLocalDisplayImage.remote` instead.

Reply:
My tests didn't fail even though this value was wrong.

Reply:
@IvaKop I guess it's better to replace all tests that involve "saved images" with E2E tests. The tests here don't seem reliable for that.

Reply:
[This test](https://github.com/CitizenLabDotCo/citizenlab/blob/3bc53601cec2dc100595ea4e91a3859e60f2ee63/front/app/components/admin/ProjectableHeaderBgUploader/ProjectableHeaderBgUploader.test.tsx) detects the same bug [in this line](https://github.com/CitizenLabDotCo/citizenlab/blob/3bc53601cec2dc100595ea4e91a3859e60f2ee63/front/app/components/admin/ProjectableHeaderBgUploader/index.tsx#L43). So, I think a unit test is still fine here.

Reply:
If it works, I would say let's keep it? No harm in having the functionality covered by both unit and e2e tests (assuming we cover this with Cypress, already).

Reply:
I'm removing the ones with "saved" images for now and adding 1 or 2 more cases to E2E tests. Partly because I was adding them in the E2E tests already (your comment about the data-cy), so it was a duplicate anyway.

Reply:
@IvaKop my code was wrong and the tests didn't tell me, while they should have:
<img width="758" alt="Screenshot 2023-01-17 at 10 42 47" src="https://user-images.githubusercontent.com/16427929/212863924-f3c8dfb1-eda3-4876-8096-fc3f19fca5ef.png">

So it means they're unreliable for saved images, right? It should not have shown, since imageShouldBeSaved was false, but it showed anyway (in the tests). Probably similar to the issues we had last week. It seems hard to reliably render the component in tests with the saved image.

Reply:
I'm confused for a moment. Will keep you updated when it's clear.

Reply:
You didn't ask me, but I'll reply! :D 

In my understanding, `not.toBeInTheDocument` is not reliable because it's always true at the beginning. So, you should either wait for some time or to use the positive check `toBeInTheDocument` to make it reliable.

Reply:
As it's done here for example https://github.com/CitizenLabDotCo/citizenlab/blob/3bc53601cec2dc100595ea4e91a3859e60f2ee63/front/app/components/admin/ProjectableHeaderBgUploader/ProjectableHeaderBgUploader.test.tsx#L34

Reply:
Hang on for a moment. There are bugs in my code, so I need to fix these first. Wrong elements are displaying.

I'll ask you again to review when I'm done :D